### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by roxanemace**
I've upgraded all the dependencies of the repository. I had to do some changes to fix new errors:

- Union types were not accepted as a value of the `QueryString` type, so I've replaced the record's value by `any`.
- The upgrade of the `jest` and `jest-fetch-mock` packages came with the `jsdom` update which breaks typings and doesn't allow anymore to overwrite `window.location.assign` (error: `Cannot assign to read only property 'assign' of object '[object Location]'`).

### CODE REVIEW

File: src/main/__tests__/apiClient.spec.ts [hunk 1]
```diff
@@ -26,8 +26,13 @@ function createServices(config = {}) {
 }
 
 beforeEach(() => {
-  window.location.assign = jest.fn()
-  window.fetch = fetchMock
+  window.fetch = fetchMock as any
+
+  Object.defineProperty(window, 'location', {
```
Comment (FrancoisDevemy5) [inline src/main/__tests__/apiClient.spec.ts:31]: Isn't it possible to define a function for this pattern?

**Review by FrancoisDevemy5 — APPROVED** [review]
Isn't it possible to define a function for this pattern?

**Comment by roxanemace** [comment]
> Isn't it possible to define a function for the `Object.defineProperty` pattern?

Tell me what you think of the new implementation :)

File: src/utils/queryString.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import map from 'lodash/map'
 
 import { camelCaseProperties, snakeCaseProperties } from './transformObjectProperties'
 
-export type QueryString = Record<string, string | string[] | number | boolean | undefined>
```
Comment (rchapel) [inline src/utils/queryString.ts:6]: I know that `string | string[] | number | boolean | undefined` is a poor typing but It still give more information than `any`.

**Review by rchapel — APPROVED** [review]

File: src/utils/queryString.ts (hunk 2 above)
Comment (roxanemace) [reply src/utils/queryString.ts]: After some more research, I've found a better solution :)

**Pull request merged**